core/auth/src/main/java/software/amazon/awssdk/auth/credentials/internal/ProfileCredentialsUtils.java

```diff
@@ -116,6 +119,11 @@ public ProfileCredentialsUtils(Profile profile, Function<String, Optional<Profil
             }
         }
 
+        if (properties.containsKey(ProfileProperty.SSO_ROLE_NAME) || properties.containsKey(ProfileProperty.SSO_ACCOUNT_ID)
+            || properties.containsKey(ProfileProperty.SSO_REGION) || properties.containsKey(ProfileProperty.SSO_START_URL)) {
+            return Optional.ofNullable(ssoProfileCredentialsProvider());
+        }
```

Comment:
Does this match the rules in the CLI? (After role_arn, before session_token, skip SSO if any of the 4 required fields aren't set, instead of raising an error?)

Reply:
This rule isn't specified in the SEP, but there is a line saying: 
`If the SSO credential provider is added to a default chain it MUST be added immediately after the WebIdentityProvider and before all other providers.` 
Even though this isn't the default providers chain, it should still follow the similar rule so I put the SSO provider here.

About skipping SSO when none of the four fields are set, there is the corresponding rule in SEP: 
`If one or more of the above configuration values is present the profile MUST be resolved by the SSO credential provider. All of the above configuration values MUST be set in the profile for the profile to be treated as a valid profile by the SSO credential provider. In the case that one or more of the above configuration values is missing from the profile the SSO credential provider MUST raise an exception that the profile is invalid.`
From my understanding, if there is one or more of the four properties is present then we need to check if all of the four are all present, and if not then we throw out an exception. However, if none of them are here then at least it means the customer isn't wanting the SSO provider so we shouldn't raise the error.

Reply:
Should this be above the `role_arn` logic, then?

Reply:
I checked with CLI team, they confirmed that they have the same order as us (SSO is after webIdentity).
```
def providers(self, profile_name, disable_env_vars=False):
        return [
            self._create_web_identity_provider(
                profile_name, disable_env_vars,
            ),
            self._create_sso_provider(profile_name),
            self._create_shared_credential_provider(profile_name),
            self._create_process_provider(profile_name),
            self._create_config_provider(profile_name),
        ]
```

Reply:
Which step do they check the `role_arn`? Right now, it looks like we're prioritizing `role_arn` higher than `sso`. I think the CLI is prioritizing it lower.

Reply:
I think `role_arn` is only used while creating the web identity credentials provider so it should be checked at the very first step. But I can check again to confirm this more accurately.

Reply:
I guess I should be asking about STS assume-role credentials, not role_arn. I think the CLI treats STS assume-role credentials lower than SSO credentials, but this code treats it higher.